californium-core/src/main/java/org/eclipse/californium/core/network/config/NetworkConfig.java

```diff
@@ -822,4 +826,24 @@ public NetworkConfig removeConfigObserver(NetworkConfigObserver observer) {
 		observers.remove(observer);
 		return this;
 	}
+
+	public <T> NetworkConfig addEndpointSpecificConfigProvider(Class<T> clazz, EndpointSpecificConfigProvider<T> endpointSpecificConfigProvider) {
+		configProviderMap.put(clazz, endpointSpecificConfigProvider);
```

Comment:
We have some other configuration stuff, which may be added. But currently they are not endpoint specific.